### ISSUE TITLE
Support 8 digits imdb titles

### ISSUE THREAD

**Issue by lashus** [tboothman/imdbphp#163]
Support 8 digits imdb titles and people

#### Description
Currently this package doesn't support 8 digits imdb titles that are working for (I guess) some time now.

#### Movies / TV-Shows / Person
tt10044952

 (http://imdb.com/title/tt10044952)

#### Type
Bug

#### Code
Only an example - this applies basically to every file.
```php
public function seasons()
    {
        if ($this->seasoncount == -1) {
            $this->getPage("Title");
            if (preg_match_all('|href="/title/tt\d{7}/episodes\?season=\d+.*?"\s*>(\d+)</a>|Ui', $this->page["Title"],
              $matches)) {
                $this->seasoncount = $matches[1][0];
            } else {
                $this->seasoncount = 0;
            }
            if (preg_match_all('|href="/title/tt\d{7}/episodes\?season\=unknown"\s*>unknown</a>|Ui',
              $this->page["Title"], $matches)) {
                $this->seasoncount += count($matches[0]);
            }
        }
        return $this->seasoncount;
    }
```

#### Expected Results / What do you want to do?
Match \d{7,8}.

#### Actual Results / What is happening?
Basically every preg match is broken.

**Comment by lashus** [comment]
PR: https://github.com/tboothman/imdbphp/pull/164

**Comment by lashus** [comment]
Closing as PR merged :)

**Comment by tboothman** [comment]
Probably worth noting if anyone looks at this that you could using imdb titles with 8 digit numbers but links to those titles from other titles or people (e.g. an actor's filmography) wouldn't have shown up.

**Comment by paxter** [comment]
Does anybody know if imdb is changing all ids from 7 digits to 8? It looks like, some ids where changed to 8 digits and other not. I'm a little bit confused and maybe it's better to wait some days before implementing fixes.

**Comment by tboothman** [comment]
They're padding to 7 digits still, and that's still the minimum for title urls but they allow over 7 digits.
For example
https://www.imdb.com/title/tt110357/ - doesn't work
https://www.imdb.com/title/tt0110357/ - (7 digits) does work (and is how it's linked to normally)
https://www.imdb.com/title/tt00110357/ - (8 digits) works too - and if you search imdb lion king you get this url too
https://www.imdb.com/title/tt000000110357/ - >8 digits works fine

**Comment by paxter** [comment]
I know that you can add unlimited zeros before the id and it's still working. But that's not the point.
If google update ids to 8 digit length (with two leading zeros, which is new), it could mean, that imdb is upgrading their whole system to 8 digit ids.
That would mean, we only need a change to fixed 8 digits length and no 7 OR 8 digits.

For example this id: https://www.imdb.com/title/tt0033852/
Works only with two leading zeros, because imdb use a minimum of 7 digits. If you remove a zero, it does not work anymore. So this could be a system change by imdb to upgrade all of their ids to 8 digits.
